src/peer_connection.cpp

```diff
@@ -1093,7 +1092,7 @@ namespace libtorrent
 		{
 			TORRENT_TRY {
 				e->on_piece_pass(index);
-			} TORRENT_CATCH (std::exception const&) {}
```

Comment:
I wonder if using just `try` and `catch` (but still consider `BOOST_NO_EXCEPTIONS`) is better.